API/controller/mealController.js

```diff
@@ -0,0 +1,114 @@
+import mealData from "../utils/mealData";
+import mealModel from "../model/mealModels";
+
+const MealController = {
+  fetchAllMeals(req, res) {
+    return res.status(200).json({
+      status: 200,
+      data: mealData
+    });
+  },
+  addAMeal(req, res) {
+    const { name, size, price, summary, imageUrl } = req.body;
+
+    const newlyCreatedMeal = {
+      id: mealData[mealData.length - 1].id + 1,
+      name,
+      size,
+      price,
+      summary,
+      imageUrl
+    };
+
+    const data = mealData.push(newlyCreatedMeal);
+    if (data) {
+      return res.status(201).json({
+        status: 201,
+        message: "New meal has been added",
+        data: [newlyCreatedMeal]
+      });
+    }
+
+    return res.status(500).json({
+      status: 500,
+      message: "something went wrong while trying to save your data"
+    });
+  },
+
+  getSingleMeal(req, res) {
+    const id = parseInt(req.params.id, 10);
+
+    mealData.map(mealData => {
+      if (mealData.id === id) {
+        return res.status(200).send({
+          success: 200,
+          message: "Meal has been retrieved successfully",
+          mealData
+        });
+      }
+    });
+    return res.status(404).send({
+      success: 404,
+      message: "meal does not exist"
+    });
+  },
+
+  deleteMeal(req, res) {
+    const error = {};
+    const id = parseInt(req.params.id, 10);
+
+    const removedIndex = mealData.findIndex(data => data.id === +id);
+    if (removedIndex === -1) {
+      error.mgs = "Oooops! no record with such Id";
+      return res.status(404).json({ status: 404, error });
+    }
+
+    mealData.splice(removedIndex, 1);
```

Comment:
Line 60 - 66 can be condensed into one arrow function by using the [filter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter) method on the array.